### PR TITLE
[WIP] Commit if last stage, or stage used in future From

### PR DESCRIPTION
**Pull request by rhatdan**
Currently we commit every stage when the last command happens.
We should not need to do this unless it is the last stage
or this stage is used in a future From command.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@nalin is that what you were thinking?

**Comment by nalind** [comment]
Unfortunately, we need those intermediate images to always be committed so that they can be found in the cache later, per https://github.com/containers/buildah/pull/1473#issuecomment-479592881.

When we conditionalize on whether or not the stage's name is the same as stage's index, we can get an incorrect result for instructions like `COPY --from=0`, which I'd rather avoid.

File: imagebuildah/build.go [hunk 1]
```diff
@@ -1538,8 +1542,22 @@ func (b *Executor) Build(ctx context.Context, stages imagebuilder.Stages) (image
...
+		// a base image later
+		forceCommit := stageIndex == len(stages)-1
+		if !forceCommit {
+			for index, value := range b.stages {
+				if value == stageExecutor {
+					continue
+				}
+				if index != stage.Name {
+					forceCommit = true
+				}
```
Comment (nalind) [inline imagebuildah/build.go]: I think that when we have multiple stages, this loop ends up always setting `forceCommit` to `true`?

**Comment by rhatdan** [comment]
@nalind patch is better https://github.com/containers/buildah/pull/1517

**Comment by nalind** [comment]
Thanks!

**Pull request closed without merging**